ARCHITECTURE.adoc

```diff
@@ -0,0 +1,100 @@
+== Architecture
+
+This document describes the high-level architecture of the Statefulset Resize Controller.
+It should give you a good impression on _how_ it works and _where_ to change things.
+
+=== Overview
+
+On a very high-level, the Statefulset Resize Controller watches changes on Statefulsets and reacts to them.
+This is the general way Kubernetes Operators work and we call this reacting to resource changes the _reconcile loop_.
+If you are not familiar with Kubernetes Operators it might be helpful to read up on them before proceeding.
+
+The controller will notice Statefulsets that should be resized, make a copy of the current volumes by creating new volumes with the requested size, delete the previous volume and then restore the original data from the previous volumes to the new volumes.
+It does this by only interacting with the Kubernetes API and has no access to other systems.
```

Comment:
Still missing a few.
```suggestion
On a very high-level, the StatefulSet Resize Controller watches changes on StatefulSets and reacts to them.
This is the general way Kubernetes Operators work and we call this reacting to resource changes the _reconcile loop_.
If you are not familiar with Kubernetes Operators it might be helpful to read up on them before proceeding.

The controller will notice StatefulSets that should be resized, make a copy of the current volumes by creating new volumes with the requested size, delete the previous volume and then restore the original data from the previous volumes to the new volumes.
It does this by only interacting with the Kubernetes API and has no access to other systems.
```

Reply:
Sorry my sleepy brain wasn't quite able to write the right find replace command..